drivers/platform/aducm3029/irq.c

```diff
@@ -0,0 +1,298 @@
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include "irq.h"
+#include "irq_extra.h"
+#include "error.h"
+#include <stdlib.h>
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+/** The number of the first external interrupt, used by NVIC */
+#define BASE_XINT_NB			(XINT_EVT0_IRQn)
+
+/** Number of interrupts controllers available */
+#define NB_INTERRUPT_CONTROLLERS	1u
+
+/** Map the interrupt ID to the ADI_XINT_EVENT associated event */
+static const uint32_t id_map_event[NB_EXT_INTERRUPTS] = {
+	ADI_XINT_EVENT_INT0, // ID 0
+	ADI_XINT_EVENT_INT1, // ID 1
+	ADI_XINT_EVENT_INT2, // ID 2
+	ADI_XINT_EVENT_INT3  // ID 3
+};
+
+/******************************************************************************/
+/***************************** Global Variables *******************************/
+/******************************************************************************/
+
+/** Bitmap with initialized interrupt */
+static uint32_t		initialized;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Call the user callback
+ * @param aducm_desc - Descriptor where the user callback is stared
+ * @param event - Event that generated the callback
+ * @param arg - Unused
+ */
+static void internal_callback(void *aducm_desc, uint32_t event, void *arg)
+{
+	struct aducm_irq_desc *desc = aducm_desc;
+
+	(void)arg;
+
+	switch (event) {
```

Comment:
Can't this switch be replaced by an if statement?
For example:
```
if (event < ADI_XINT_MAX_INT_NUM && desc->irq_handler[event])
    desc->irq_handler[event]((void *)event);
```

Reply:
done